packages/flame/lib/src/components/component.dart

```diff
@@ -384,33 +462,14 @@ class Component {
   /// See FlameGame.changePriority instead.
   void changePriorityWithoutResorting(int priority) => _priority = priority;
 
-  /// Prepares the [Component] to be added to a [parent], and if there is an
-  /// ancestor that is a [FlameGame] that game will do necessary preparations
-  /// for this component.
-  /// If there are no parents that are a [Game] false will be returned and this
-  /// will run again once an ancestor or the component itself is added to a
-  /// [Game].
-  @protected
-  @mustCallSuper
-  void prepare(Component parent) {
-    _parent = parent;
-
-    final parentGame = findParent<FlameGame>();
-    if (parentGame == null) {
-      isPrepared = false;
-    } else if (!isPrepared) {
-      assert(
-        parentGame.hasLayout,
-        '"prepare/add" called before the game is ready. '
-        'Did you try to access it on the Game constructor? '
-        'Use the "onLoad" or "onMount" method instead.',
-      );
-      parentGame.prepareComponent(this);
-
-      debugMode |= parent.debugMode;
-      isPrepared = true;
-    }
-  }
+  /// Component at the root of the component tree. This node serves as the
+  /// central place for resolving lifecycle event queues, and also facilitates
+  /// `onGameResize` events. This variable MUST be initialized before components
+  /// may be combined into a tree.
+  ///
+  /// Usually, [FlameGame] is the root of the component tree, and it declares
+  /// itself as such automatically.
+  static ComponentTreeRoot? root;
```

Comment:
How will this work when we have multiple different games on the screen at the same time?

Reply:
Each game is a component, and they all can be mounted to a single "master" component and shown on the screen simultaneously.
Or do you have in mind some specific example that you're worrying about?

Reply:
Yeah, on that case it will work, but since this is a static variable, and there will be only one value during the entire app runtime. So I imagine that if we have two games running on different game widgets, I wonder if this will work fine

Reply:
Yes, since this is a static variable, there could be some difficulties. It does help though that `ComponentTreeRoot` is designed in such a way that it doesn't _have_ to be physically the root-most component of the tree. So, in theory, multiple independent component trees can share the same "root". We have the "game-in-game" example to demonstrate this.

Ultimately, this use case seems to be purely hypothetical, though. I asked Lukas about this, and neither he nor I were able to come up with a single game example where this could possibly be needed. Can you think of any?

Reply:
The situation Erick is talking about is different than the one we were talking about, in Erick's sitation there are multiple `GameWidget`s in the same Fluttet widget tree, not multiple games in a component tree. That usecase is much more common since people don't only use Flame for full blown games, but it could be for only having sprites animated somewhere in their normal app for example.

I think this could be a real problem because one `GameWidget` could be disposed before the other and if that one has the root I guess things could blow up?

Reply:
Well, I imagine in such a situation one would have to be careful to create a single root at the top level, and then not to reassign it when creating internal `Game`s. We could even have an extra parameter `root=false` to `FlameGame` that will prevent it from acquiring the root.

@spydon Would this be ok, or did you have something else in mind?

Reply:
> Well, I imagine in such a situation one would have to be careful to create a single root at the top level.

Imho users that create apps just using small parts of Flame shouldn't have to consider the widget lifecycle differently just because they happen to have `GameWidget`s in their widget tree.

> We could even have an extra parameter root=false to FlameGame that will prevent it from acquiring the root.

I don't think it would help, the developer doesn't necessarily know in which order the widgets will live.

> did you have something else in mind?

I don't, but I think we have to come up with another way.

Reply:
> but I think we have to come up with another way.

Do you want to do it within this PR, or leave it as a follow-up issue for the future?

Reply:
We could have a function like `Flame.initializeMultiGameSupport()` that needs to be called at the start of the application, and which creates and stores a single "master" game instance, while prohibiting all subsequent games from overwriting it. Would this approach work?

Reply:
It may work, but I don't like that approach, Flame games should be self contained and they shouldn't have global stuff IMO

Reply:
I believe the main principle was that "common tasks should be easy, and uncommon feasible". The overwhelmingly common case here is to use Flame engine to build games, using FlameGame class or its derivatives. When one builds a game, there is always only one game (it is impossible to play two games at a time). Thus, in practice, a Game is always a singleton, whether we recognize it officially as such or not.

It is only when one tries to use FlameGame class to build non-games, this becomes a concern. But these use-cases are extraordinarily rare. I don't think we ever mention them in documentation or tutorials. So IMHO it's ok for them to be special cases.

Reply:
But this is still a common case already today, so we can't overlook it. `Game` is not a singleton, and should not be a singleton. It is pretty much one of our first points every time we present Flame; that it is just another widget.